cy - refactored e2e tests (fixtures)

artshllk opened the pull request:
## :unicorn: Problem
We needed to refactor the Cypress tests to improve readability and maintainability, particularly in how fixtures were loaded across the test suite. The previous approach involved loading each fixture manually within the `beforeEach` block, leading to code duplication and decreased clarity.

## :robot: Solution
I refactored the Cypress test suite by introducing a more concise method to handle fixture loading. Instead of calling `cy.task("db:fixture", ...)` for each fixture separately, I grouped the fixtures into an array and used a loop to load them all at once. This reduces redundancy, improves readability, and makes future modifications easier (e.g., adding new fixtures).

The refactor only affects the way fixtures are loaded, without impacting the overall test functionality. This should help streamline future test maintenance and updates.

## :rainbow: Notes
- The fixtures array contains all the necessary fixtures and is processed in the `beforeEach` hook to ensure that they are loaded before each test.
- No changes were made to the actual test logic, only the setup phase for better readability.

## :100: How to test
1. Run the Cypress tests as usual.
2. Ensure that all tests relying on database fixtures run without issues and that the fixtures are loaded correctly for each test case.
3. Review the fixture-loading behavior to verify that it's functioning as expected across all test environments.

artshllk committed:
cy - refactored e2e tests (fixtures)

artshllk committed:
revert describe desc

VincentHardouin commented on high-level-tests/e2e/cypress/integration/pix-app/a11y-authenticated-1.test.js:
```diff
@@ -4,39 +4,41 @@ describe("a11y", () => {
-    cy.task("db:fixture", "answers");
-    cy.task("db:fixture", "knowledge-elements");
-  });
+  const authenticatedPages = [
+    "/mes-tutos/enregistres",
+    "/mes-tutos/recommandes",
+    "/mon-compte/informations-personnelles",
+    "/mon-compte/langue",
+    "/mon-compte/methodes-de-connexion",
+    "/plan-du-site",
+  ];
```
Some pages are missing in this array

fael-b commented on high-level-tests/e2e/cypress/integration/pix-app/a11y-authenticated-1.test.js:
```diff
@@ -47,13 +49,12 @@ describe("a11y", () => {
         viewports.forEach(({ width, height }) => {
           cy.viewport(width, height);
-          cy.checkA11yAndShowViolations({ skipFailures, url });
+          cy.checkA11yAndShowViolations({ skipFailures: false, url });
```
`skipFailures` was set to `true` for some urls, notably `/assessments/fake-assessment`

artshllk committed:
improvements

artshllk commented:
Sorry for the small mistakes earlier! I’ve addressed the feedback and made the necessary updates. Everything should look good now, but please let me know if there’s anything else I need to fix or adjust. I was just aiming to do a small refactor to clean things up a bit.

artshllk committed:
Merge branch 'dev' into cy-refactor

yaf commented:
Hello,

Thanks a lot for your rework.
We will close this old PR. 
We don't want to change our e2e fixtures structuration.

Perhaps next time !
